nemo/collections/vlm/neva/data/llava_next_energon.py

```diff
@@ -0,0 +1,161 @@
+from typing import Dict, List
+
+import torch
+from megatron.energon import VQASample, batch_list, batch_pad_stack
+from torch.nn.utils.rnn import pad_sequence
+
+from nemo.collections.multimodal.data.energon.config import ImageTextRawBatch, ImageTextSample, MultiModalSampleConfig
+from nemo.collections.multimodal.data.energon.sample_encoder import SampleEncoder, VQASampleEncoder
+from nemo.collections.multimodal.data.energon.task_encoder import MultiModalTaskEncoder
+from nemo.utils import logging
+
+
+class LlavaNextTextSample(ImageTextSample):
+    num_media_tiles: int = 0
+
+
+@dataclass
+class LlavaNextTextRawBatch(ImageTextRawBatch):
+    num_media_tiles: List[int] = field(default_factory=list)
+
+
+class LlavaNextSampleEncoder(VQASampleEncoder):
+    def __init__(self, tokenizer, image_processor, multimodal_sample_config=MultiModalSampleConfig()):
+        """
+        Initialize the LlavaNextSampleEncoder, inherited from VQASampleEncoder for multimodal samples
+        focused on VQA-style data to support LLaVANeXT
+
+        Parameters:
+        tokenizer (Tokenizer): The HF tokenizer used for processing text.
+        image_processor (ImageProcessor): The HF image processor used for preprocessing images.
+        multimodal_sample_config (MultiModalSampleConfig, optional): Configuration object for multimodal samples.
+            Defaults to MultiModalSampleConfig().
+        """
+        super().__init__(tokenizer, image_processor, multimodal_sample_config)
+
+    def process_image(self, image):
+        """
+        Process and prepare an image sample for encoding.
+
+        This method preprocesses the image using the HF image_processor, converting it to
+        a tensor.
+
+        Parameters:
+        image: The input image to be processed.
+
+        Returns:
+        torch.Tensor: The processed image tensor.
+        """
+        image_array = self.image_processor.preprocess(image, return_tensors='pt', do_rescale=False)['pixel_values'][0]
+        return image_array
+
+    def encode(self, input_sample: VQASample, output_sample: LlavaNextTextSample):
+        """
+        Encode a single sample into a format suitable for model input.
+
+        This method prepares the conversation prompt, tokenizes it, and processes
+        the associated image. It fills the output sample with tokens, labels, loss mask,
+        and other required fields for multimodal processing.
+
+        Parameters:
+        input_sample (VQASample): The input VQA sample containing an image and conversation text.
+        output_sample (LlavaNextTextSample): The output sample structure where encoded results are stored.
+
+        Returns:
+        LlavaNextTextSample: The encoded output sample, containing processed tokens, labels,
+            images, loss masks, and metadata.
+        """
+        conversation_prompt = self.apply_prompt_template(input_sample)
+        logging.debug(f"task encoder encode_sample conversation_prompt {conversation_prompt}")
```

Comment:
can you keep the logging consistent with nemo/collections/vlm/mllama/data/sample_encoder.py
         logging.debug(f"[Energon] task encoder encode_sample conversation_prompt {conversation_prompt}")
you put energon tag before